source/core/serial_solid.cc

```diff
@@ -95,47 +99,54 @@ SerialSolid<dim, spacedim>::map_solid_in_background_triangulation(
+      // Gather the amount of vertices per cell in the solid cells once to have
+      // a static memory allocation of the vector. This prevents reallocating
+      // the vector dynamically using push_back if the solid triangulation
+      // has many triangles
+      auto                         temporary_solid_cell = solid_tria->begin();
+      std::vector<Point<spacedim>> triangle(temporary_solid_cell->n_vertices());
+
+      // Calculate distance from cell center to solid_cell
+      for (const auto &background_cell : background_tr.active_cell_iterators())
         {
-          // Calculate the characteristic size of the background cell
-          const double bg_cell_length = background_cell->diameter();
+          // If the cell is owned by the processor
+          if (background_cell->is_locally_owned())
+            {
+              // Calculate the characteristic size of the background cell
+              const double bg_cell_length = background_cell->diameter();
 
-          // Calculate the center of the cell
-          Point<spacedim> bg_cell_center = background_cell->center();
+              // Calculate the center of the cell
+              Point<spacedim> bg_cell_center = background_cell->center();
 
-          // Calculate distance from center of the cell to triangle
-          for (auto &solid_cell : solid_tria->active_cell_iterators())
-            {
-              // Gather triangle vertices
-              for (unsigned int v = 0; v < solid_cell->n_vertices(); ++v)
+              // Calculate distance from center of the cell to triangle
+              for (auto &solid_cell : solid_tria->active_cell_iterators())
                 {
-                  triangle[v] = solid_cell->vertex(v);
-                }
+                  // Gather triangle vertices
+                  for (unsigned int v = 0; v < solid_cell->n_vertices(); ++v)
+                    {
+                      triangle[v] = solid_cell->vertex(v);
+                    }
 
-              // Calculate distance between triangle and center
-              const double distance =
-                LetheGridTools::find_point_triangle_distance(triangle,
-                                                             bg_cell_center);
+                  // Calculate distance between triangle and center
+                  const double distance =
+                    LetheGridTools::find_point_triangle_distance(
+                      triangle, bg_cell_center);
 
-              if (distance < bg_cell_length)
-                {
-                  mapped_solid.push_back(
-                    std::make_pair(background_cell, solid_cell));
+                  if (distance < bg_cell_length)
+                    {
+                      mapped_solid.push_back(
+                        std::make_pair(background_cell, solid_cell));
+                    }
                 }
             }
         }
     }
+  else if constexpr (dim == 3)
+    {
+    }
```

Comment:
Can you add a throw statement here or something because I don't want us to get the impression that this works.

Reply:
I suggested a warning message, but yeah, maybe a throw is better

Reply:
I would also used a warning, otherwise all the date structure are there for nothing. 

Also, it's not documented, so no one should be using it.

Reply:
It's going to be a Throw, otherwise we need to add the ``pcout`` as a parameter.